packages/serverpod/lib/src/server/endpoint_dispatch.dart

```diff
@@ -52,6 +52,41 @@ abstract class EndpointDispatch {
     return connector;
   }
 
+  Future<(MethodConnector, Map<String, dynamic>)> tryGetEndpointMethod({
+    required Session session,
+    required String endpointPath,
+    required String methodName,
+    required String body,
+    required SerializationManager serializationManager,
+    Map<String, dynamic> additionalParameters = const {},
+  }) async {
+    var connector = getConnectorByName(endpointPath);
+    if (connector == null) {
+      throw EndpointNotFoundException('Endpoint $endpointPath not found');
+    }
+
+    var authenticationFailedResult = await canUserAccessEndpoint(
+      () => session.authenticated,
+      connector.endpoint.requireLogin,
+      connector.endpoint.requiredScopes,
+    );
+    if (authenticationFailedResult != null) {
+      throw NotAuthorizedException(authenticationFailedResult);
+    }
+
+    var method = connector.methodConnectors[methodName];
+    if (method is! MethodConnector) {
+      throw MethodNotFoundException(
+          'Method "$methodName" not found in endpoint: $endpointPath');
+    }
+
+    var paramMap = parseParameters(
+        body.isEmpty ? null : body, method.params, serializationManager,
```

Comment:
Question: Would it make sense to have this `empty` check on the inside of the parseParameters method instead? And then strictly take in a typed `String`.

Reply:
Definitely 👍

Reply:
Refactored into different helpers.